turbinia/evidence.py

```diff
@@ -300,19 +341,37 @@ class GoogleCloudDiskRawEmbedded(GoogleCloudDisk):
 
   Attributes:
     embedded_path: The path of the raw disk image inside the Persistent Disk
+    _clouddisk_mount_path(str): the path where the attached disk is mounted.
   """
 
   def __init__(self, embedded_path=None, *args, **kwargs):
     """Initialization for Google Cloud Disk."""
     self.embedded_path = embedded_path
     super(GoogleCloudDiskRawEmbedded, self).__init__(*args, **kwargs)
 
+    self._clouddisk_mount_path = None
+
   def _preprocess(self):
-    self.local_path = google_cloud.PreprocessAttachDisk(self.disk_name)
-    self.loopdevice_path = mount_local.PreprocessLosetup(self.local_path)
-    self.mount_path = mount_local.PreprocessMountDisk(
+    self._attached_path = google_cloud.PreprocessAttachDisk(self.disk_name)
+    self.loopdevice_path = mount_local.PreprocessLosetup(self._attached_path)
+    self._clouddisk_mount_path = mount_local.PreprocessMountDisk(
         self.loopdevice_path, self.mount_partition)
-    self.local_path = os.path.join(self.mount_path, self.embedded_path)
+
+    # If the mounting of the embedded image step fails, at least local_path
+    # still points to the image.
+    self.local_path = os.path.join(
+        self._clouddisk_mount_path, self.embedded_path)
+
+    try:
+      # We now try to mount the image pointed by self.embedded_path
+      self.mount_path = mount_local.PreprocessMountDisk(
+          self.embedded_path, self.mount_partition)
```

Comment:
The embedded_path here is the relative path inside the outer cloud disk.  I think what we want here is what is assigned to self.local_path at this point (The joining of the mount path of the outer cloud disk and the inner relative embedded_path).